pynitrokey/start/upgrade_by_passwd.py

```diff
@@ -427,116 +465,116 @@ def start_update(regnual, gnuk, default_password, password, wait_e, keyno, verbo
-                    local_print('*** Device returns "Invalid PIN" error.'
-                          + ' ' + str_factory_reset)
-                break
+                    local_critical("- device returns: 'Invalid PIN' error",
+                                   "- please retry with correct PIN", e)
         except Exception as e:
-            # unknown error, bail
-            local_print('*** Found unexpected error: {}'.format(str(e)))
-            break
+            local_critical("unexpected error", e)
 
     if not update_done:
-        local_print()
-        local_print('*** Could not proceed with the update. Please execute one or all of the following and try again:\n'
-              '- reinsert device to the USB slot;\n'
-              '- run factory-reset on the device;\n'
-              '- close other applications, that possibly could use it (e.g. scdaemon, pcscd).\n')
-        exit(1)
+        local_critical("",
+            "Could not proceed with the update",
+            "Please execute one or all of the following and try again:",
+            "- re-insert device to the USB slot",
+            "- run factory-reset on the device",
+            "- close other applications, which could use it (e.g., scdaemon, pcscd)")
 
     dev_strings_upgraded = None
     takes_long_time = False
-    local_print('Currently connected device strings (after upgrade):')
+    local_print("Currently connected device strings (after upgrade):")
     for i in range(TIME_DETECT_DEVICE_AFTER_UPDATE_S):
         if i > TIME_DETECT_DEVICE_AFTER_UPDATE_LONG_S:
             if not takes_long_time:
-                local_print('\n*** Please reinsert device to the USB slot')
+                local_print("", "Please reinsert device to the USB slot")
                 takes_long_time = True
         time.sleep(1)
         dev_strings_upgraded = get_devices()
         if len(dev_strings_upgraded) > 0:
             local_print()
             print_device(dev_strings_upgraded[0])
             break
-        local_print('.', end='', flush=True)
+        local_print(".", end="", flush=True)
 
     if not dev_strings_upgraded:
-        local_print()
-        local_print('Could not connect to the device. '
-              'It should be working fine though after power cycle - please reinsert device to '
-              'USB slot and test it.')
-        local_print('Device could be removed from the USB slot.')
-    logger.debug('Final device strings: {}'.format(dev_strings_upgraded))
-    logger.debug('Finishing session {}'.format(datetime.now()))
-    local_print('Log saved to: {}'.format(UPGRADE_LOG_FN))
+        local_print("",
+            "could not connect to the device - might be due to a failed update",
+            "please re-insert the device, check version using:",
+            "$ nitropy start list")
+
+    local_print(f"device can now be safely removed from the USB slot",
+                f"final device strings: {dev_strings_upgraded}")
```

Comment:
`dev_strings_upgraded` is not meant to be visible for user

Reply:
next PR